cilium/bpf_metadata.cc

```diff
@@ -561,17 +567,36 @@ Network::FilterStatus Instance::onAccept(Network::ListenerFilterCallbacks& cb) {
   Network::Socket::appendOptions(socket_options,
                                  Network::SocketOptionFactory::buildReusePortOptions());
 
-  // keep alive (SO_KEEPALIVE, TCP_KEEPINTVL, TCP_KEEPIDLE)
-  Network::Socket::appendOptions(
-      socket_options,
-      Network::SocketOptionFactory::buildTcpKeepaliveOptions(Envoy::Network::TcpKeepaliveConfig{
-          .keepalive_probes_ = absl::nullopt, // not setting TCP_KEEPCNT
-          .keepalive_time_ = 5 * 60,          // 5 min
-          .keepalive_interval_ = 5 * 60,      // 5 min
-      }));
-
+  // Adding SocketOptions to the downstream socket. The function `setOption` is NOT executed
+  // on the downstream socket itself - it's executed later on the corresponding upstream socket!
   socket.addOptions(socket_options);
 
+  // set keep alive socket options on accepted connection socket
+  // (SO_KEEPALIVE, TCP_KEEPINTVL, TCP_KEEPIDLE)
+  int keepalive = true;
```

Comment:
see original commit: https://github.com/cilium/proxy/pull/1115/commits/13d6409dbc1e168a002cad87c4e2c6a41c38f3f7

it's not a 1:1 revert - error handling is different and fails in case socket option can't be applied.

Reply:
This is fine :-)